docs/deployment-and-operations/state-bootstrap.md

```diff
@@ -0,0 +1,137 @@
+
+{% highlight java %}
+public class MyStateBootstrapFunction implements StateBootstrapFunction {
+
+	@Persisted
+	private PersistedValue<MyState> state = PersistedValue.of("my-state", MyState.class);
+
+	@Override
+	public void bootstrap(Context context, Object input) {
+		state.set(extractStateFromInput(input));
+	}
+ }
+{% endhighlight %}
+
+Assume that this bootstrap function was provided for function type ``MyFunctionType``, and the id of the bootstrap function instance was ``id-13``. 
+The function writes persisted state of name ``my-state`` using the given bootstrap data. 
+After restoring a Stateful Functions application from the savepoint generated using this bootstrap function, the stateful function instance with address ``(MyFunctionType, id-13)`` will already have state values available under state name `my-state`.
+
+## Creating A Savepoint
+
+Savepoints are created by defining certain metadata, such as max parallelism and state backend.
+The default state backend is [RocksDB](https://ci.apache.org/projects/flink/flink-docs-stable/ops/state/state_backends.html#the-rocksdbstatebackend).
+
+{% highlight java %}
+int maxParallelism = 128;
+StatefulFunctionsSavepointCreator newSavepoint = new StatefulFunctionsSavepointCreator(maxParallelism);
+{% endhighlight %}
+
+Each input data set is registered in the savepoint creator with a [router]({{ site.baseurl }}/io-module/index.html#router) that routes each record to zero or more function instances.
+You may then register any number of function types to the savepoint creator, similar to how functions are registered within a stateful functions module.
+Finally, specify an output location for the resulting savepoint.
+
+{% highlight java %}
+// Read data from a file, database, or other location
+final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+final DataSet<Tuple2<String, Integer>> userSeenCounts = env.fromElements(
+	Tuple2.of("foo", 4), Tuple2.of("bar", 3), Tuple2.of("joe", 2));
+
+// Register the dataset with a router
+newSavepoint.withBootstrapData(userSeenCounts, MyStateBootstrapFunctionRouter::new);
+
+// Register a bootstrap function to process the records
+newSavepoint.withStateBootstrapFunctionProvider(
+		new FunctionType("apache", "my-function"),
+		ignored -> new MyStateBootstrapFunction());
+
+newSavepoint.write("file:///savepoint/path/");
+
+env.execute();
+{% endhighlight %}
+
+For full details of how to use Flink's DataSet api, please check the official [documentation](https://ci.apache.org/projects/flink/flink-docs-stable/dev/batch/).
+
+## Deployment
+
+After creating a new savpepoint, it can be used to provide the initial state for a Stateful Functions application.
+
+<div class="codetabs" markdown="1">
+<div data-lang="Image Deployment" markdown="1">
+When deploying based on an image, pass the ``-s`` command to Flink [JobMaster](https://ci.apache.org/projects/flink/flink-docs-stable/concepts/glossary.html#flink-master) image.
```

Comment:
```suggestion
When deploying based on an image, pass the ``-s`` command to the Flink [JobMaster](https://ci.apache.org/projects/flink/flink-docs-stable/concepts/glossary.html#flink-master) image.
```

Reply:
We don’t call it JobMaster in statefun when deploying from an image.
We call it simply master.
Let’s be consistent with this. 
We can put Flink JobMaster in parentheses

Reply:
how about just "pass the -s command to the master image"?